pkg/scheduler/eventhandlers.go

```diff
@@ -157,59 +157,39 @@ func (sched *Scheduler) deleteNodeFromCache(obj interface{}) {
 }
 
 func (sched *Scheduler) onCSINodeAdd(obj interface{}) {
-	csiNode, ok := obj.(*storagev1beta1.CSINode)
+	_, ok := obj.(*storagev1beta1.CSINode)
 	if !ok {
 		klog.Errorf("cannot convert to *storagev1beta1.CSINode: %v", obj)
 		return
 	}
 
-	if err := sched.config.SchedulerCache.AddCSINode(csiNode); err != nil {
-		klog.Errorf("scheduler cache AddCSINode failed: %v", err)
-	}
-
 	sched.config.SchedulingQueue.MoveAllToActiveQueue()
```

Comment:
I don't think this is correct. The CSINode isn't set at all, so there will be no volume limits.

Reply:
@bertinatto 
Do you have recommendation on what onCSINodeUpdate should do ?

Reply:
The SchedulerCache.AddCSINode call may return error (which was ignored).
It seems this func doesn't depend on the successful addition to cache.